src/symbol/sym.rs

```diff
@@ -0,0 +1,48 @@
+//! Implementation of a symbol table entry that will automatically
+//! demangle rustc names.
+
+use addr2line::demangle_auto;
+use std::borrow::Cow;
+use std::ops::{Deref, DerefMut};
+
+/// A symbol table entry.
+#[derive(Clone, Debug)]
+pub struct Symbol<'data> {
+    symbol: object::Symbol<'data>,
+}
+
+impl<'data> Symbol<'data> {
+    /// Returns the demangled name if this symbol has a name.
+    #[inline]
+    pub fn name(&self) -> Option<String> {
```

Comment:
```suggestion
    pub fn demangled_name(&self) -> Option<String> {
```

Then `orig_name` is not necessary anymore.